lib/cb/utils/api.rb

```diff
@@ -57,10 +57,14 @@ def cb_delete(path, options = {}, &block)
       def execute_http_request(http_method, uri, path, options = {}, &block)
```

Comment:
This method is starting to grow kinda lengthy. I see a possible small change that could happen - breaking into 2 methods, one that starts the timing and sets up stuff like `api_caller` and eventing (at least begin, maybe both), and another method that handles the wrapped execution and returns the response. It's more of a nice to have, style type thing though. I can read through it the way it is now, and the tests make it very clear that the logic here is correct.